packages/sbom-tools/src/commands/scan-node-js.ts

```diff
@@ -100,6 +72,58 @@ async function formatVuln(
   };
 }
 
+async function fetchScore(vulnId: string, nodeVuln: NodeVuln) {
+  try {
+    const cves = await Promise.all(
+      nodeVuln.cve.map((cve) =>
+        fetch(
+          `https://services.nvd.nist.gov/rest/json/cves/2.0?cveId=${cve}`
+        ).then((res) => res.json())
+      )
+    );
+
+    const getBestCvssMetricScore = (
+      cvssMetrics: {
+        type: 'Primary' | 'Secondary';
+        cvssData: { baseScore: number };
+      }[]
+    ) => {
+      return (
+        cvssMetrics.find((m) => m.type === 'Primary')?.cvssData?.baseScore ??
+        cvssMetrics.find((m) => m.type === 'Secondary')?.cvssData?.baseScore
+      );
+    };
+
+    const allCvss: (number | undefined)[] = cves.map(
+      (cve) =>
+        getBestCvssMetricScore(
+          cve?.vulnerabilities[0]?.cve?.metrics?.cvssMetricV31 ?? []
+        ) ??
+        getBestCvssMetricScore(
+          cve?.vulnerabilities[0]?.cve?.metrics?.cvssMetricV30 ?? []
+        ) ??
+        getBestCvssMetricScore(
+          cve?.vulnerabilities[0]?.cve?.metrics?.cvssMetricV2 ?? []
+        )
+    );
+
+    const knownCvss: number[] = [];
+
+    for (const cvss of allCvss) {
+      if (typeof cvss === 'number') {
+        knownCvss.push(cvss);
+      }
+    }
+
+    return knownCvss.length ? Math.max(...knownCvss) : undefined;
+  } catch (e) {
```

Comment:
The tiniest nit: I'd only try/catch around the `await Promise.all()` because that's where network /3rd party errors can happen. Now you're also doing try/catch around all our own potential runtime programming errors like "null pointer exceptions" (ie. reading something of undefined or whatever).

Although in this case that's probably not the end of the world because it would continue with undefined which (it seems) would be treated as the highest severity, so probably OK.

Reply:
ie. I'd usually prefer runtime programming errors to blow up the application just like it would outside of this try/catch.

Reply:
that's a very good point, i'll update it